.github/workflows/e2e-sample-apps-tests.yml

```diff
@@ -0,0 +1,254 @@
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout: release
+      - name: Prepare build scripts
+        run: cd ${{ github.workspace }}/release && yarn && yarn build
+      - name: Get build dependencies
+        uses: actions/github-script@v7
+        id: dependencies
+        with:
+          script: | # js
+            const {
+              getBuildRequirements,
+              getVersionFromReleaseBranch,
+            } = require('${{ github.workspace }}/release/dist/index.cjs');
+            const targetBranchName = '${{ github.base_ref || github.ref }}';
+
+            const version = getVersionFromReleaseBranch(targetBranchName);
+            const requirements = getBuildRequirements(version);
+
+            return {
+              java_version: requirements.java,
+              node_version: requirements.node,
+            };
+
+  get-sample-apps-data:
+    if: |
+      !cancelled()
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    outputs:
+      branch_name: ${{ fromJson(steps.build-sample-apps-data.outputs.result).branch_name }}
+      matrix: ${{ toJSON(fromJson(steps.build-sample-apps-data.outputs.result).matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Prepare build scripts
+        run: cd ${{ github.workspace }}/release && yarn && yarn build
+      - name: Build sample app compatibility data
+        uses: actions/github-script@v7
+        id: build-sample-apps-data
+        with:
+          script: | # js
+            const {
+              isReleaseBranch,
+              getMajorVersionNumberFromReleaseBranch,
+            } = require('${{ github.workspace }}/release/dist/index.cjs');
+
+            const testSuites = JSON.parse(process.env.TEST_SUITES);
+            const testSuitesToRunInMasterOnly = JSON.parse(process.env.TEST_SUITES_TO_RUN_IN_MASTER_ONLY);
+
+            const ref = '${{ github.base_ref || github.ref }}';
+            const branchName = ref.replace(/^refs\/heads\//, "");
+
+            const isMainBranchRef = branchName === 'master';
+            const isReleaseBranchRef = isReleaseBranch(branchName);
+
+            if (isMainBranchRef) {
+              return {
+                branch_name: 'main',
+                matrix: testSuites,
+              };
+            }
+
+            if (isReleaseBranchRef) {
+              const supportedReleasesCount = Number(${{ env.SUPPORTED_RELEASES_COUNT }});
+
+              const currentReleaseVersion = Number(${{ vars.CURRENT_VERSION }});
+              const currentBranchVersion = Number(getMajorVersionNumberFromReleaseBranch(branchName));
+
+              const shouldRunTestsForVersion = (currentReleaseVersion - currentBranchVersion) <= (supportedReleasesCount - 1);
+              const releaseBranchTestSuites = testSuites.filter(testSuite => !testSuitesToRunInMasterOnly.includes(testSuite));
+
+              return {
+                // branch_name: branchName,
+                branch_name: 'main',
```

Comment:
So the merge is soon, so temporarily, I set the `main` branch here, because Sample Apps don't have release branches yet